src/plugins/no_data_page/server/routes/index.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { IRouter, KibanaResponseFactory, RequestHandlerContext } from '@kbn/core/server';
+import { NO_DATA_API_PATHS } from '../../common';
+import type { HasApiKeysApiResponse } from '../../common/types';
+
+export const getHasApiKeysRoute = (router: IRouter) => {
+  router.get(
+    {
+      path: NO_DATA_API_PATHS.internal.hasApiKeys,
+      validate: {},
+    },
+    async function handler(ctx: RequestHandlerContext, _req: unknown, res: KibanaResponseFactory) {
+      const esClient = (await ctx.core).elasticsearch.client;
+
+      // get API key info
+      const result = await esClient.asCurrentUser.security.getApiKey({ owner: true });
+      const { api_keys: apiKeys } = result;
```

Comment:
cc @legrego this gets a list of security keys using a method of the ES client. While I would prefer to go through the Security plugin contract in doing this, I didn't see any methods that would work nicely here. LMK if there are other things to think about.

Reply:
> While I would prefer to go through the Security plugin contract in doing this, I didn't see any methods that would work nicely here

I was thinking we would introduce whatever functionality we required into the security plugin's API Key Service.

>  LMK if there are other things to think about.

1. The current user may not have privileges to call this API. Make sure you are handling this lack of authorization correctly (i.e. don't return the raw ES error to the client, don't not handle the error, etc.)
2. API Keys may not be enabled. This isn't a problem for serverless, but [it's possible to disable this service](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-settings.html#api-key-service-settings) for on prem. The security plugin [already has logic](https://github.com/elastic/kibana/blob/4dbcda37c5a782d6a48e13fc83763bb04f15c831/x-pack/plugins/security/server/authentication/api_keys/api_keys.ts#L86-L89) to detect this.

Reply:
Creating a separate PR: https://github.com/elastic/kibana/pull/172884